src/generator/state-machine-to-intermediate.js

```diff
@@ -154,18 +154,37 @@ const getIntermediateFormOf = ({left, right, operator, out}, isSecondarySignal)
     }
 };
 
+// Helper function to generate next available numeric label
+function generateNextLabel(states) {
+    let max = 0;
+    states.forEach(({state, statements}) => {
+        if(typeof state === 'number') {
+            const end = statements
+                .map(({start, operations}) => start + operations.length)
+                .reduce((a, b) => Math.max(a, b), 0); // Last statement is always a branch, so no need to check past end
+            if(end > max) {
+                max = end + 1;
+            }
+        }
+    });
+    return Math.ceil(max / 10) * 10; // Round to the nex multiple of 10
+}
+
```

Comment:
Will this work if the programmer mixes numeric labels with alphanumeric ones? Is that something we should support?

Reply:
Yes, I will add a test for this